docs/petr4spec/Petr4-spec.mdk

```diff
@@ -142,509 +149,2979 @@ p4grammar {
+|   | $\vert$ | $\int$ | $\qquad \text{integers}$ |
+|   | $\vert$ | $n_w$ | $\qquad \text{fixed-width signed integers}$ |
+|   | $\vert$ | $b_w$ | $\qquad \text{fixed-width unsigned integers}$ |
+|   | $\vert$ | $\name$ | $\qquad \text{names (variables)}$ |
+|   | $\vert$ | $\arrayAccess {\exp_1} {\exp_2}$ | $\qquad \text{array accesses}$ |
+|   | $\vert$ | $\bitStringAccess {\exp_1} {\exp_2} {\exp_3}$ | $\qquad \text{bitstring slices}$ |
+|   | $\vert$ | $\list \exps$ | $\qquad \text{lists}$ |
+|   | $\vert$ | $\records \exp$ | $\qquad \text{records}$ |
+|   | $\vert$ | $\unaryOp \exp$ | $\qquad \text{unary operations}$ |
+|   | $\vert$ | $\exp_1 \binOp \exp_2$ | $\qquad \text{binary operations}$ |
+|   | $\vert$ | $\cast \baseTyp \exp$ | $\qquad \text{casts}$ |
+|   | $\vert$ | $\typMem \typVar \field$ | $\qquad \text{type members}$ |
+|   | $\vert$ | $\errMem \field$ | $\qquad \text{error members}$ |
+|   | $\vert$ | $\expMem \exp \field$ | $\qquad \text{expression members}$ |
+|   | $\vert$ | $\ternary {\exp_1} {\exp_2} {\exp_3}$ | $\qquad \text{conditionals}$ |
+|   | $\vert$ | $\funcCall \exp {\baseTyps} \args$ | $\qquad \text{function calls}$ |
+|   | $\vert$ | $\instantiation {\baseTyp} {\args}$ | $\qquad \text{anonymous instantiation}$ |
+|   | $\vert$ | $\mask {\exp_1} {\exp_2}$ | $\qquad \text{bit masks}$ |
+|   | $\vert$ | $\range {\exp_1} {\exp_2}$ | $\qquad \text{ranges}$ |
+
+**Labels:**
+
+|     |      |     |     |
+|~~~~:|:~~~~:|:~~~~|:~~~~|
+| $\lbl$ | $::=$ | $\defLbl$ | $\qquad \text{default}$ |
+|   | $\vert$ | $\str$ | $\qquad \text{string labels}$ |
+
+**Switchs:**
+
+|     |      |     |     |
+|~~~~:|:~~~~:|:~~~~|:~~~~|
+| $\switchCase$ | $::=$ | $\actionCase \stmts$ | $\qquad \text{labeled block}$ |
+|   | $\vert$ | $\lbl$ | $\qquad \text{fall through with label}$ |
+
+
+**Statements:**
+
+|     |      |     |     |
+|~~~~:|:~~~~:|:~~~~|:~~~~|
+| $\stmt$ | $::=$ | $\methodCall \exp \baseTyps \args$ | $\qquad \text{method call}$ |
+|   | $\vert$ | $\assign {\exp_1} {\exp_2}$ | $\qquad \text{assignment}$ |
+|   | $\vert$ | $\dirApp \baseTyp \args$ | $\qquad \text{direct application}$ |
+|   | $\vert$ | $\ifthen \exp \stmt$ | $\qquad \text{if then}$ |
+|   | $\vert$ | $\ifthenelse \exp {\stmt_1} {\stmt_2}$ | $\qquad \text{if then else}$ |
+|   | $\vert$ | $\block \stmts$ | $\qquad \text{sequencing}$ |
+|   | $\vert$ | $\exit$ | $\qquad \text{exit}$ |
+|   | $\vert$ | $\noop$ | $\qquad \text{noop}$ |
+|   | $\vert$ | $\return \exp$ | $\qquad \text{return}$ |
+|   | $\vert$ | $\retNothing$ | $\qquad \text{return nothing}$ |
+|   | $\vert$ | $\switch \exp \switchCases$ | $\qquad \text{switch}$ |
+|   | $\vert$ | $\dcl$ | $\qquad \text{declaration}$ |
+
+**Parser states:**
+
+|     |      |     |     |
+|~~~~:|:~~~~:|:~~~~|:~~~~|
+| $\nxt$ | $::=$ | $\str$ | $\qquad \text{next state}$ |
+| $\pmatch$ | $::=$ | $\defMatch$ | $\qquad \text{default match}$ |
+|   | $\vert$ | $\dontcareMatch$ | $\qquad \text{don't care match}$ |
```

Comment:
Same comment about how $\dontcareMatch$ is rendered as above.

Reply:
- [ ] we want it to render to `dontcare`?